pkg/messaging/nats/pubsub.go

```diff
@@ -36,7 +36,7 @@ type PubSub interface {
 }
 
 type pubsub struct {
-	conn          *broker.Conn
+	publisher
```

Comment:
Needs variable name

Reply:
@drasko I suggested using the composition for reasons explained [here](https://github.com/mainflux/mainflux/pull/1562#discussion_r868436254).